Add transient validation methods and corresponding tests

pbearne opened the pull request:
A new function named 'valid_transient' has been introduced to verify if a transient and its value are valid, also by checking if they have not expired. Sites with transient data can use the 'valid_site_transient' function to perform similar checks. Along with these functions, PHPUnit test cases have been added to ensure these functions perform as expected.


Trac ticket: https://core.trac.wordpress.org/ticket/37040

Commit:
Add transient validation methods and corresponding tests

A new function named 'valid_transient' has been introduced to verify if a transient and its value are valid, also by checking if they have not expired. Sites with transient data can use the 'valid_site_transient' function to perform similar checks. Along with these functions, PHPUnit test cases have been added to ensure these functions perform as expected.

Commit:
whitespace

Commit:
whitespace

swissspidy reviewed (commented):
The docblocks need to be fixed according to the coding standards (change order, use correct version number)

Also, I suggest using `is_valid_` as the prefix

Commit:
Rename transient validation function names

The function names 'valid_transient' and 'valid_site_transient' have been renamed to 'is_valid_transient' and 'is_valid_site_transient' respectively. This change has been reflected across the codebase, ensuring tests and code coverage annotations are updated appropriately. These function names now better reflect their purpose and adhere to common coding conventions.

Commit:
Rename transient validation functions and update tests

The function names "valid_transient" and "valid_site_transient" have been renamed to "is_valid_transient" and "is_valid_site_transient" respectively for better readability and to follow naming conventions. All references to these functions in the code and tests have been updated accordingly.

Commit:
Fixed formatting in function parameters

Small adjustments made to improve code readability, complying with WordPress coding standards. Both 'is_valid_transient' and 'is_valid_site_transient' function parameters' formatting have been updated, ensuring consistency throughout the file.

mukeshpanchal27 commented on src/wp-includes/option.php:
```diff
@@ -1385,6 +1385,39 @@ function delete_transient( $transient ) {
+ * If the transient does not exist, does not have a value, or has expired,
+ * then the return value will be false.
+ *
+ * @since 6.5.0
```
Should be next version `6.6.0`

Commit:
Update version number in option.php comments

The version numbers in the comments of the option.php file have been updated. This change adjusts the version number from 6.5.0 to 6.6.0 in two places, reflecting the most recent version where the described transient behaviors occur.

felixarntz reviewed (changes requested):
@pbearne This looks like a good start for option-based transients. But per https://core.trac.wordpress.org/ticket/37040#comment:11, it will need to be expanded to support a persistent object cache.
